master/pkg/model/environment_config.go

```diff
@@ -47,7 +50,9 @@ func (r *RuntimeItem) UnmarshalJSON(data []byte) error {
 	var plain string
 	if err := json.Unmarshal(data, &plain); err == nil {
 		r.CPU = plain
-		r.GPU = plain
+		r.ROCM = plain
+		r.CUDA = plain
+		r.GPU = r.CUDA
 		return nil
 	}
 	type DefaultParser RuntimeItem
```

Comment:
Either have a temporary struct here that includes the deprecated GPU field or and have some logic to shim it into a modern, or just parse it into a `map[string]string`.  But this is where the `gpu` field from old configs should be stripped and should never make it further in to the go code.

(well, here and in the copy inside `pkg/schemas/expconf` and in RuntimeItems and in the copy of RuntimeItems in `pkg/schemas/expconf`)

Reply:
got it done through a shim with two-stage parsing of sorts.

Reply:
works for me